src/datasets/datasets.controller.ts

```diff
@@ -941,13 +942,27 @@ export class DatasetsController {
     description: "Return new value of the dataset",
   })
   async appendToArrayField(
+    @Req() request: Request,
     @Param("pid") id: string,
     @Query("fieldName") fieldName: string,
     @Query("data") data: string,
   ): Promise<DatasetClass | null> {
+    const loggedInUser: JWTUser = request.user as JWTUser;
+
+    const datasetToUpdate = await this.datasetsService.findOne({
+      where: { pid: id },
+    });
+
+    // NOTE: Not sure if email is the best way to do this but since we don't have userId or something like that email is acceptable.
+    const couldUpdateDataset =
+      datasetToUpdate?.ownerEmail === loggedInUser.email;
     // $addToSet is necessary to append to the field and not overwrite
     // $each is necessary as data is an array of values
 
+    if (!couldUpdateDataset) {
+      throw new ForbiddenException();
+    }
+
```

Comment:
In order to allow the update on the dataset, we should use the casl factory.
The rules is the following:
- if the user logged in belongs to the group that owns the dataset, he/she is allowed to update the dataset.
- if the user belongs to any of the admin groups, he/she can update the dataset.

Reply:
Ahaaa... I see now but do we have these rules set up already or I need to fix that?

Reply:
@nitrosx I fixed it and now it uses casl to decide if user is able to update something or not.